### PR TITLE
Oversight routes, JSON and MySQL implementations

### PR DESCRIPTION
**Pull request by jacobkrantz**
New routes:
- `/api/v1/dispensers/single/:dispenserID`
- `/api/v1/dispensers/all`
- `/api/v1/dispensers/:name (updated)`
- `/api/v1/prescribers/single/:prescriberID`
- `/api/v1/prescribers/all`
- `/api/v1/prescribers/:name`
- `/api/v1/prescriptions/prescriber/:prescriberID`

**Tests for each new route**

All route implementations are in JSON for now. Will be migrated to MySQL in the future.

Is the information about a prescriber correct, or am I missing fields? (answer: all good now!)

This PR now also includes MySQL integration for dispenser and prescriber searching.

### CODE REVIEW

**Comment by andrewjyang** [comment]
@jacobkrantz 
i would appreciate routes for almost every field a prescriber and dispenser has.
need routes
for dispensers:
- phone number
- location
for prescribers: 
- phone number
- i think prescribers should have a first name and last name (similar to patients)
- also i think prescribers should have an office location that users can also search for (similar to dispenser location)
thank you <3

**Comment by jacobkrantz** [comment]
@mdulin2 @andrewjyang Ready for full review.

**Comment by jacobkrantz** [comment]
@andrewjyang @mdulin2 MySQL migration of dispenser and prescriber data complete. Tests pass for both MySQL connection and dummy data. Dummy data kept so that Travis tests still pass. Please take a look!

**Comment by jacobkrantz** [comment]
@andrewjyang You promised Friday, but this is still open and not reviewed.

**Comment by andrewjyang** [comment]
@jacobkrantz 
issue: 
- when using the dispenser name query (with a substring 'cost' for example), it increments the dispenser id by 1 for some unknown reason.
curl "http://localhost:5000/api/v1/dispensers/all"
>>
 [{"dispenserID":0,"name":"Walgreens","phone":15553336464,"location":"8840 E Dragontail Rd"},{"dispenserID":1,"name":"Costco","phone":15551106464,"location":"8705 N Colchuck Ln"},{"dispenserID":2,"name":"Safeway","phone":15559376464,"location":"8453 E Argonaut Dr"},{"dispenserID":3,"name":"Yokes","phone":15552126464,"location":"8364 McClellan Pl"},{"dispenserID":4,"name":"Rite Aid","phone":15553456464,"location":"9415 Stuart Rd"},{"dispenserID":5,"name":"Walgreens","phone":15555676464,"location":"8605 Sherpa Rd"}]

but,
curl "http://localhost:5000/api/v1/dispensers?name=cost"
>> 
[{"dispenserID":2,"name":"Costco","location":"8705 N Colchuck Ln","phone":15551106464}]

the dispenser Costco should have the dispenserID of 1 not 2.

File: WebPhapp/backend/server.js [hunk 1]
```diff
@@ -961,15 +1054,83 @@ app.get('/api/v1/dispensers/prescriptions/open/:dispenserID', (req, res) => {
...
+    An api endpoint that returns all dispensers.
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/dispensers/all"
+    To be used in Axois call:
+        .get("/api/v1/dispensers/all")
+Returns:
+    [
+        {
+            prescriberID (int),
```
Comment (andrewjyang) [inline WebPhapp/backend/server.js]: nit: need to update comments in name queries to include first and last name fields :)

File: WebPhapp/backend/server.js [hunk 2]
```diff
@@ -981,30 +1142,207 @@ Returns:
...
+    An api endpoint that returns all prescribers.
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/prescribers/all"
+    To be used in Axois call:
+        .get("/api/v1/prescribers/all")
+Returns:
+    [
+        {
+            prescriberID (int),
```
Comment (andrewjyang) [inline WebPhapp/backend/server.js]: nit: also need to update comments in name queries to include first and last name fields :)

File: WebPhapp/backend/server.js [hunk 3]
```diff
@@ -981,30 +1142,207 @@ Returns:
...
+        res.status(400).send(false);
+    }
+    else {
+        prescriber = prescribers[0]
+        console.log('/api/v1/prescribers/single/: returning prescriber with ID ' + prescriberID.toString());
+        res.status(200).send(prescriber);
+    }
+});
+
+/*
```
Comment (andrewjyang) [inline WebPhapp/backend/server.js]: issue: name queries increment prescriber id by 1

File: WebPhapp/backend/server.js [hunk 4]
```diff
@@ -961,15 +1054,83 @@ app.get('/api/v1/dispensers/prescriptions/open/:dispenserID', (req, res) => {
...
+*/
+app.get('/api/v1/dispensers/all', (req,res) => { // auth.checkAuth([Role.Government]),
+    var dispensers = readJsonFileSync(
+        __dirname + '/' + "dummy_data/dispensers.json").dispensers;
+
+    console.log('/api/v1/dispensers/all: returning ' + dispensers.length + ' dispensers.');
+    res.status(200).send(dispensers);    
+});
+
 /*
```
Comment (andrewjyang) [inline WebPhapp/backend/server.js]: issue: name query increment dispenser id by 1

File: WebPhapp/backend/server.js (hunk 1 above)
Comment (jacobkrantz) [reply WebPhapp/backend/server.js]: dispenser info includes name, dispenserID, location, and phone. Name is not first and last. Let me know if we are not on the same page there.

**Comment by jacobkrantz** [comment]
@andrewjyang back to you!

File: WebPhapp/backend/server.js (hunk 1 above)
Comment (andrewjyang) [reply WebPhapp/backend/server.js]: you right

**Pull request merged**